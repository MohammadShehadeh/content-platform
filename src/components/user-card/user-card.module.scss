.email {
  display: block;
  color: hsl(var(--muted-foreground));
  font-weight: var(--font-weight-medium);
  font-size: var(--font-size-sm);
}

.link {
  gap: var(--space-sm);
}
