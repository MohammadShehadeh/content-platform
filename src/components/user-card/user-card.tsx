import { ArrowRight } from 'lucide-react';
import Link from 'next/link';

import { Button } from '@/components/shared/button';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/shared/card';
import type { User } from '@/services/users';

import styles from './user-card.module.scss';

type UserCardProps = Pick<User, 'name' | 'email' | 'id'> & {
  className?: string;
};

export const UserCard = ({ name, email, id, className }: UserCardProps) => {
  return (
    <Card className={className}>
      <CardHeader>
        <CardTitle>
          {name}
          <span className={styles.email}>{email}</span>
        </CardTitle>
      </CardHeader>
      <CardContent>
        <Button variant="link" size="sm" asChild>
          <Link href={`/users/${id}`} className={styles.link}>
            View User
            <ArrowRight width={16} height={16} />
          </Link>
        </Button>
      </CardContent>
    </Card>
  );
};
