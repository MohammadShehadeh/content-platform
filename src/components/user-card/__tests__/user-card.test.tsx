import { render, screen } from '@testing-library/react';

import { UserCard } from '@/components/user-card';

describe('UserCard', () => {
  it('should render user information correctly', () => {
    render(<UserCard name="John Doe" email="john@example.com" id={1} />);

    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('john@example.com')).toBeInTheDocument();
    expect(screen.getByText('View User')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/users/1');
  });

  it('should render with custom className', () => {
    const { container } = render(
      <UserCard name="Jane Smith" email="jane@example.com" id={2} className="custom-card" />
    );
    expect(container.firstChild).toHaveClass('custom-card');
  });

  it('should match snapshot', () => {
    const { container } = render(<UserCard name="John Doe" email="john@example.com" id={1} />);
    expect(container).toMatchSnapshot();
  });
});
