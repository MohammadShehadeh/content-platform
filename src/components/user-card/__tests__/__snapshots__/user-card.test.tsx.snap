// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard matches snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        John Doe
        <span
          class="email"
        >
          john@example.com
        </span>
      </h3>
    </div>
    <div
      class="content"
    >
      <a
        class="button link sm link"
        href="/users/1"
      >
        View User
        <svg
          class="lucide lucide-arrow-right"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="m12 5 7 7-7 7"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`UserCard should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <h3
        class="title"
      >
        John Doe
        <span
          class="email"
        >
          john@example.com
        </span>
      </h3>
    </div>
    <div
      class="content"
    >
      <a
        class="button link sm link"
        href="/users/1"
      >
        View User
        <svg
          class="lucide lucide-arrow-right"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
          <path
            d="m12 5 7 7-7 7"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
